### PR TITLE
engine: GetSubsystemsStatus fix

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Fix GetSubsystemsStatus bug that was suppressing comm manager output via gctcli.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by thrasher- — APPROVED** [review]
Good catch! 👀

**Review by gloriousCode — APPROVED** [review]
tACK! What kind of person would make such a mistake??? 👀 Thanks for the pickup!
```
{
 "subsystems_status": {
  "communications": false,
  "data_history_manager": true,
  "database": true,
  "deprecated_rpc": true,
  "dispatch": true,
  "exchange_syncer": true,
  "gctscript": false,
  "grpc": true,
  "grpc_proxy": true,
  "internet_monitor": true,
  "ntp_timekeeper": false,
  "orders": true,
  "portfolio": true,
  "websocket_rpc": false
 }
}
```

File: engine/connection_manager.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func (m *connectionManager) Start() error {
 
 // Stop stops the connection manager
 func (m *connectionManager) Stop() error {
-	if m == nil {
+	if m == nil || m.conn == nil {
```
Comment (gloriousCode) [inline engine/connection_manager.go:74]: I think `m.conn == nil` should be its own if to highlight setup process issues, similar to `IsOnline`

File: engine/helpers.go [hunk 2]
```diff
@@ -66,29 +69,40 @@ type RPCEndpoint struct {
...
-		ListenAddr: "http://" + Bot.Config.RemoteControl.DeprecatedRPC.ListenAddress,
-	}
-	endpoints[WebsocketName] = RPCEndpoint{
-		Started:    Bot.Settings.EnableWebsocketRPC,
-		ListenAddr: "ws://" + Bot.Config.RemoteControl.WebsocketRPC.ListenAddress,
-	}
-	return endpoints
+func (bot *Engine) GetRPCEndpoints() (map[string]RPCEndpoint, error) {
+	if bot.Config == nil {
+		return nil, errConfigIsNil
```
Comment (gloriousCode) [inline engine/helpers.go:74]: We have `errNilConfig` already

**Review by gloriousCode — CHANGES_REQUESTED** [review]
_Two tiny nits_

**Review by thrasher- — APPROVED** [review]
tACK, thanks for expanding the test coverage too!

```sh
$ ./gctcli getinfo
{
 "uptime": "48.837085s",
 "available_exchanges": 27,
 "enabled_exchanges": 27,
 "default_forex_provider": "ExchangeRateHost",
 "default_fiat_currency": "USD",
 "subsystem_status": {
  "communications": false,
  "data_history_manager": false,
  "database": false,
  "deprecated_rpc": true,
  "dispatch": true,
  "exchange_syncer": true,
  "gctscript": true,
  "grpc": true,
  "grpc_proxy": false,
  "internet_monitor": true,
  "ntp_timekeeper": false,
  "orders": true,
  "portfolio": true,
  "websocket_rpc": true
 },
 "rpc_endpoints": {
  "deprecated_rpc": {
   "started": true,
   "listen_address": "http://localhost:9050"
  },
  "grpc": {
   "started": true,
   "listen_address": "grpc://localhost:9052"
  },
  "grpc_proxy": {
   "listen_address": "http://localhost:9053"
  },
  "websocket_rpc": {
   "started": true,
   "listen_address": "ws://localhost:9051"
  }
 }
}
```

**Pull request merged**